JavaScriptSPA/authPopup.js

```diff
@@ -14,10 +14,23 @@ function signIn() {
         
     }).catch(function (error) {
       console.log(error);
+
+      // Error handling
+      if (error.errorMessage) {
+        // Check for forgot password error
+        // Learn more about AAD error codes at https://docs.microsoft.com/en-us/azure/active-directory/develop/reference-aadsts-error-codes
+        if (error.errorMessage.indexOf("AADB2C90118") > -1) {
+          myMSALObj.loginPopup(b2cPolicies.authorities.forgotPassword)
```

Comment:
@derisen, when I implement `myMSALObj.loginPopup(b2cPolicies.authorities.forgotPassword)` it is blocked by the popup blocker (Chrome 85). I suspect that is because the last click was in the signIn popup which has been closed after the user clicked "forgot password". Is there some way you handle that elsewhere in the code?

Reply:
@Ruud-Zuiderlicht you should be able to use the redirect flow instead. In the `JavaScriptSPA/index.html`:

```html
  <!-- replace next line with authRedirect.js if you would like to use the redirect flow -->
  <!-- <script type="text/javascript" src="./authRedirect.js"></script>   -->
  <script type="text/javascript" src="./authPopup.js"></script>
```

Just switch the files and you should be fine.

Reply:
That's not always as easy as "replacing popup by redirect". For example an SPA will loose state and therefor needs to add even more logic to recover after the user returns. Do you have this popup flow working without being blocked in any browser?

Reply:
Popups can always be blocked by the browser. By default, browsers will usually allow popups that are opened as the direct result of user interaction, such as a button click.